Attribute of typedef dropped, leads to failure of extraction.

Minimal example to check this:
```c
typedef unsigned short int sa_family_t;
typedef unsigned int __socklen_t;
typedef __socklen_t socklen_t;
typedef int ares_socket_t;
typedef socklen_t ares_socklen_t;

struct sockaddr {
  sa_family_t sa_family;
  char sa_data[14];
};

typedef union {
  struct sockaddr *__restrict __sockaddr__;
} __SOCKADDR_ARG __attribute__((__transparent_union__));

extern int getsockname(int __fd, __SOCKADDR_ARG __addr,
                       socklen_t *__restrict __len)
    __attribute__((__nothrow__));

int main() {
  struct sockaddr *src_addr;
  ares_socket_t sock;
  ares_socklen_t len;
  getsockname(sock, src_addr, &len);
}
```
extract with the following command
```bash
clang-extract exp.c  -DCE_EXTRACT_FUNCTIONS=main -DCE_DUMP_PASSES                                        05:51:01
exp.c:34:21: error: passing 'struct sockaddr *' to parameter of incompatible type '__SOCKADDR_ARG'
   34 |   getsockname(sock, src_addr, &len);
      |                     ^~~~~~~~
exp.c:25:49: note: passing argument to parameter '__addr' here
   25 | extern int getsockname(int __fd, __SOCKADDR_ARG __addr,
      |                                                 ^

Error on pass: ClosurePass
```
The original source should have been compiled with clang, examine the dumpped pass, the `__attribute__((__transparent_union__))` is dropped:
```
/** clang-extract: from exp.c:1:1  */
typedef unsigned short int sa_family_t;

/** clang-extract: from exp.c:2:1  */
typedef unsigned int __socklen_t;

/** clang-extract: from exp.c:3:1  */
typedef __socklen_t socklen_t;

/** clang-extract: from exp.c:4:1  */
typedef int ares_socket_t;

/** clang-extract: from exp.c:5:1  */
typedef socklen_t ares_socklen_t;

/** clang-extract: from exp.c:10:1  */
struct sockaddr;

/** clang-extract: from exp.c:12:1  */
typedef union {
  struct sockaddr *__restrict __sockaddr__;
} __SOCKADDR_ARG;

/** clang-extract: from exp.c:16:1  */
extern int getsockname(int __fd, __SOCKADDR_ARG __addr,
                       socklen_t *__restrict __len)
    __attribute__((__nothrow__));

/** clang-extract: from exp.c:20:1  */
int main() {
  struct sockaddr *src_addr;
  ares_socket_t sock;
  ares_socklen_t len;
  getsockname(sock, src_addr, &len);
}
```

--- a/libcextract/PrettyPrint.cpp
+++ b/libcextract/PrettyPrint.cpp
@@ -390,6 +390,19 @@ SourceLocation PrettyPrint::Get_Expanded_Loc(Decl *decl)
       }
     }
   }
+
+  /* In case this decl is a TypedefNameDecl, we also need to check for the
+     struct definition for extra attributes.  See small/attr-10.c testscase
+     for an example where this happens.  */
+  if (TypedefNameDecl *typedecl = dyn_cast<TypedefNameDecl>(decl)) {
+    if (TagDecl *tag = typedecl->getAnonDeclWithTypedefName()) {
+      SourceLocation tag_furthest = PrettyPrint::Get_Expanded_Loc(tag);
+      if (Is_Before(furthest, tag_furthest)) {
+        furthest = tag_furthest;
+      }
+    }
+  }
+
   return furthest;
 }
 

--- a/testsuite/small/attr-10.c
+++ b/testsuite/small/attr-10.c
@@ -1,0 +1,29 @@
+/* { dg-options "-DCE_EXTRACT_FUNCTIONS=main -DCE_NO_EXTERNALIZATION" }*/
+
+typedef unsigned short int sa_family_t;
+typedef unsigned int __socklen_t;
+typedef __socklen_t socklen_t;
+typedef int ares_socket_t;
+typedef socklen_t ares_socklen_t;
+
+struct sockaddr {
+  sa_family_t sa_family;
+  char sa_data[14];
+};
+
+typedef union {
+  struct sockaddr *__restrict __sockaddr__;
+} __SOCKADDR_ARG __attribute__((__transparent_union__));
+
+extern int getsockname(int __fd, __SOCKADDR_ARG __addr,
+                       socklen_t *__restrict __len)
+    __attribute__((__nothrow__));
+
+int main(void) {
+  struct sockaddr *src_addr;
+  ares_socket_t sock;
+  ares_socklen_t len;
+  getsockname(sock, src_addr, &len);
+}
+
+/* { dg-final { scan-tree-dump "__attribute__\(\(__transparent_union__\)\)" } } */

